lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+                    quarantine(qdir, idxctx.logger, tb)
+                    continue
+                controller_path = rp.parent
+                controller = controller_path.name
+                archive_path = controller_path.parent
+                if str(archive_path) != str(ARCHIVE_rp):
+                    idxctx.logger.warning(
+                        "For tar ball {}, original home is not {}", tb, ARCHIVE_rp
+                    )
+                    quarantine(qdir, idxctx.logger, tb)
+                    continue
+                if not Path(f"{rp}.md5").is_file():
+                    idxctx.logger.warning("Missing .md5 file for {}", tb)
+                    quarantine(qdir, idxctx.logger, tb)
+                    # Audit should pick up missing .md5 file in ARCHIVE directory.
+                    continue
+                try:
+                    # get size
+                    size = rp.stat().st_size
+                except OSError:
+                    idxctx.logger.warning("Could not fetch tar ball size for {}", tb)
+                    quarantine(qdir, idxctx.logger, tb)
+                    # Audit should pick up missing .md5 file in ARCHIVE directory.
+                    continue
+                else:
+                    tarballs.append((size, controller, tb))
+        except SigTermException:
+            # Re-raise a SIGTERM to avoid it being lumped in with general
+            # exception handling below.
+            raise
+        except Exception:
+            idxctx.logger.exception(
+                "Unexpected error encountered generating list"
+                " of tar balls to process"
+            )
+            # tuple to return the status and return value
+            return (12, [])
+        else:
+            if not tarballs:
+                idxctx.logger.info("No tar balls found that need processing")
+                return (0, [])
+
+        return (0, sorted(tarballs))
+
+    def process_tb(self, tarballs):
+        """Process Tarballs For Indexing and creates report
+
+            "tarballs" - List of tarball, it is the second value of
+                the tuple returned by collect_tb() """
+
+        # We always process the smallest tar balls first.
+        idxctx = self.idxctx
+
+        # At this point, tarballs contains a list of tar balls sorted by size
+        # that were available as symlinks in the various 'linksrc' directories.
+        idxctx.logger.debug("Preparing to index {:d} tar balls", len(tarballs))
+
+        try:
+            # Now that we are ready to begin the actual indexing step, ensure we
+            # have the proper index templates in place.
+            idxctx.logger.debug("update_templates [start]")
+            idxctx.templates.update_templates(idxctx.es)
+        except TemplateError as e:
+            idxctx.logger.error("update_templates [end], error {}", repr(e))
+            res = 9
+        except SigTermException:
```

Comment:
Do we need to handle SigTermException here?

Reply:
@ndokos, see the comment on lines 153 and 154. If there is a better way to word that comment to make this requirement clear, let me know.